### PR TITLE
Allow configuration of initial list sorting

### PR DESCRIPTION
**Pull request by sleidig**
and small cleanups (see commit messages); consider merging as rebase instead of squashed

### CODE REVIEW

File: src/app/core/entity-components/entity-list/EntityListConfig.ts [hunk 1]
```diff
@@ -37,6 +38,12 @@ export interface EntityListConfig {
    */
   filters?: FilterConfig[];
 
+  /**
+   * Optional initial sort order.
+   * Default is to sort by the first column.
+   */
+  sortByDefault?: Sort;
```
Comment (TheSlimvReal) [inline src/app/core/entity-components/entity-list/EntityListConfig.ts:45]: I would understand this as a question: "Sort By default? Yes/No". How about `defaultSort`?

File: src/app/core/ui/search/search.component.ts [hunk 2]
```diff
@@ -46,10 +46,14 @@ export class SearchComponent {
     private entities: EntityRegistry
   ) {
     this.results = this.formControl.valueChanges.pipe(
-      debounceTime(this.INPUT_DEBOUNCE_TIME_MS),
+      tap(x => console.log("starting", new Date().toISOString())),
```
Comment (TheSlimvReal) [inline src/app/core/ui/search/search.component.ts:49]:
```suggestion
```

File: src/app/core/ui/search/search.component.ts [hunk 3]
```diff
@@ -46,10 +46,14 @@ export class SearchComponent {
...
+      debounceTime(SearchComponent.INPUT_DEBOUNCE_TIME_MS),
+      tap(x => console.log("debounced", new Date().toISOString())),
```
Comment (TheSlimvReal) [inline src/app/core/ui/search/search.component.ts:51]:
```suggestion
```

File: src/app/core/ui/search/search.component.ts [hunk 4]
```diff
@@ -46,10 +46,14 @@ export class SearchComponent {
...
       skipUntil(this.createSearchIndex()),
+      tap(x => console.log("index ready", new Date().toISOString())),
```
Comment (TheSlimvReal) [inline src/app/core/ui/search/search.component.ts:53]:
```suggestion
```

File: src/app/core/ui/search/search.component.ts [hunk 5]
```diff
@@ -46,10 +46,14 @@ export class SearchComponent {
...
       tap((next) => (this.state = this.updateState(next))),
-      concatMap((next: string) => this.searchResults(next))
+      concatMap((next: string) => this.searchResults(next)),
+    tap(x => console.log("done", new Date().toISOString())),
```
Comment (TheSlimvReal) [inline src/app/core/ui/search/search.component.ts:56]:
```suggestion
```

**Review by TheSlimvReal — APPROVED** [review]
Just a small renaming idea to make it clearer. Besides that looks very good.

File: src/app/core/entity-components/entity-list/EntityListConfig.ts (hunk 1 above)
Comment (sleidig) [reply src/app/core/entity-components/entity-list/EntityListConfig.ts:45]: fair point :-)

**Pull request merged**